### PR TITLE
feat[WIP]: Home page

### PR DESCRIPTION
**Pull request by saadahmadhere**
This PR focuses on:
1. Adding home page.
2. Adding categories link.
3. Adding footer.

Please note that you don't need to review the footer code as it has already been reviewed earlier.

### CODE REVIEW

File: src/Components/Categories/CategoryItem.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  const fetchCategories = async () => {
+    try {
+      const categoriesData = await axios.get("/api/categories");
+      dispatch({ type: "CATEGORIES", payload: categoriesData.data.categories });
+      console.log(categoriesData.data.categories);
+    } catch (error) {
+      console.error(error);
+    } finally {
+      console.log(categories);
```
Comment (shivampandey0) [inline src/Components/Categories/CategoryItem.jsx]: remove `console.log()`

File: src/Components/Categories/CategoryItem.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }
+  };
+
+  useEffect(() => {
+    if (categories.length === 0) fetchCategories();
+  }, []);
+
+  console.log(categories);
```
Comment (shivampandey0) [inline src/Components/Categories/CategoryItem.jsx]: remove

**Review by abhij1607 — COMMENTED** [review]
Clean Code | | Good Practices followed | | Good Atomic file & functions

File: src/Components/Categories/CategoryItem.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  return (
+    <div className="category_items flex">
+      {categories.map((categoryItem) => (
+        <div className="category_product" key={categoryItem.id}>
+          <img
+            src={categoryItem.image}
+            alt="Earphones"
```
Comment (shivampandey0) [inline src/Components/Categories/CategoryItem.jsx]: `alt` should be dynamic for different categories. Currently it's fixed to `Earphones` only

File: src/Screens/HomeScreen/HomeScreen.jsx [hunk 4]
```diff
@@ -1,10 +1,29 @@
...
   return (
-    <div>
+    <>
       <Navbar />
-    </div>
+      <div className="home_banner ">
+        <h1
+          className="h1 gray_subtitle mb_4 txt_center"
+          style={{ fontSize: "4rem" }}>
+          End Of Reason SALE
```
Comment (shivampandey0) [inline src/Screens/HomeScreen/HomeScreen.jsx:15]: > End Of Reason SALE

shouldn't it be  `Season` ?

File: src/Screens/ProductsScreen/ProductsScreen.jsx [hunk 5]
```diff
@@ -5,26 +5,28 @@ import { useState, useEffect } from "react";
...
-        console.log(err);
-      } finally {
-        setLoader(false);
-      }
-    })();
+    if (products.length === 0) fetchData();
   }, []);
 
-  console.log(`loader: ${loader}`);
   console.log(products);
```
Comment (shivampandey0) [inline src/Screens/ProductsScreen/ProductsScreen.jsx]: remove `console.log()`

**Review by shivampandey0 — COMMENTED** [review]
Code looks clean with the use of components in pages. Use of context to avoid the prop drilling.
- Added a few suggestions.

File: src/Components/Categories/CategoryItem.jsx (hunk 1 above)
Comment (saadahmadhere) [reply src/Components/Categories/CategoryItem.jsx]: done

File: src/Components/Categories/CategoryItem.jsx (hunk 2 above)
Comment (saadahmadhere) [reply src/Components/Categories/CategoryItem.jsx]: done

File: src/Screens/ProductsScreen/ProductsScreen.jsx (hunk 5 above)
Comment (saadahmadhere) [reply src/Screens/ProductsScreen/ProductsScreen.jsx]: done

File: src/Screens/HomeScreen/HomeScreen.jsx (hunk 4 above)
Comment (saadahmadhere) [reply src/Screens/HomeScreen/HomeScreen.jsx:15]: It's reason only

File: src/Components/Categories/CategoryItem.jsx (hunk 3 above)
Comment (saadahmadhere) [reply src/Components/Categories/CategoryItem.jsx]: done

**Pull request merged**